packages/eslint-plugin/src/rules/type-annotation-spacing.ts

```diff
@@ -177,6 +177,22 @@ export default util.createRule<Options, MessageIds>({
       const { before, after } = getRules(ruleSet, typeAnnotation);
 
       if (type === ':' && previousToken.value === '?') {
+        if (sourceCode.isSpaceBetween!(previousToken, punctuatorTokenStart)) {
+          context.report({
+            node: punctuatorTokenStart,
+            messageId: 'unexpectedSpaceBefore',
```

Comment:
I think we could add a custom message here for additional clarity.
Something like `Unexpected space between the {{previousToken}} and the {{type}}`.